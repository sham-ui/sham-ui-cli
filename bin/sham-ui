#!/usr/bin/env node

require( 'commander' )
    .version( require( '../package' ).version )
    .usage( '<command> [options]' )
    .command( 'new', 'generate a new project from a template' ).alias( 'n' )
    .command( 'upgrade', 'upgrade existed project from a template' ).alias( 'u' )
    .command( 'generate', 'generate a new component' ).alias( 'g' )
    .command( 'collect', 'collect filters & directives' ).alias( 'c' )
    .command( 'destroy', 'destroy component' ).alias( 'd' )
    .parse( process.argv );