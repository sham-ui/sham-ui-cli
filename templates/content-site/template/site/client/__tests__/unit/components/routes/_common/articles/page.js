import * as directives from 'sham-ui-directives';
import { createDI } from 'sham-ui';
import renderer from 'sham-ui-test-helpers';
//eslint-disable-next-line max-len
import RoutesCommonArticlesPage  from '../../../../../../src/components/routes/_common/articles/page.sfc';

it( 'renders correctly', async() => {
    const DI = createDI();

    DI.bind( 'title', {
        change() {}
    } );

    DI.bind( 'router', {
        storage: {
            params: {
                page: 2
            }
        },
        generate: () => '/'
    } );

    const meta = renderer( RoutesCommonArticlesPage, {
        DI,
        directives,
        loadData: () => Promise.resolve( {
            articles: [],
            meta: {
                total: 0,
                limit: 20
            }
        } )
    } );

    await new Promise( resolve => setImmediate( resolve ) );

    expect( meta.toJSON() ).toMatchSnapshot();
} );
