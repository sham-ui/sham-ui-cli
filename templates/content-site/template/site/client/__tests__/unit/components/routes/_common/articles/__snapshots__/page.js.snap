// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "Options": Object {
    "articles": Array [],
    "dataLoaded": true,
    "errors": Array [],
    "getPageTitle": [Function],
    "limit": 20,
    "loadData": [Function],
    "offset": 0,
    "page": 2,
    "pathName": "",
    "queryString": "",
    "total": 0,
  },
  "html": "
<div data-test-page-body=\\"\\">
  <section class=\\"section pt-55\\">
    <!--0-->
    <!--1-->
    <div class=\\"container-fluid\\">
      <div class=\\"row\\">
        <!--2-->
        <div class=\\"col-lg-12\\">
          <div class=\\"pagination mt--10\\">
            <ul class=\\"list-inline\\">
              <!--0-->
              <li>
                <a data-test-page=\\"1\\" class=\\"\\" href=\\"/\\">1</a>
              </li>
              <!--1-->
            </ul>
          </div>
        </div>
      </div>
    </div>
  </section>
  <!--0-->
</div>
<!--0-->
<!--0-->
",
}
`;
